models/workitem_repository_blackbox_test.go

```diff
@@ -0,0 +1,80 @@
+package models_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/criteria"
+	"github.com/almighty/almighty-core/models"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateLoadDeleteListWI(t *testing.T) {
+	doWithTransaction(t, func(ts *models.GormTransactionSupport) {
```

Comment:
> The idea that you just have to look at a test to know what is broken is a red herring in most cases

Not from my experience.

>  I don't want to copy/paste the setup code x times just to satisfy an abstract notion of niceness.

Don't you have something like `beforeEach` or "extact to funciton" to easily fix it? 

As I'm not really part of the core team I won't continue this discussion further, but I truly believe that this is not the way how the code should be written at all. It will get hard to maintain later to start with